### DISCUSSION TITLE
Stay where you were Without being Sent to homepage

### DISCUSSION THREAD

**Discussion by RealXinOfficial** [ScratchAddons/ScratchAddons#7496] [Ideas]
You were signed out but were in the MyStuff or Other links. You sign back in and goes to the Homepage and you dont like that. a Sugesston to add will be Once you sign back In does Not send to homepage and stay where you were before without taking too many time.

**Comment by DNin01** [comment]
This type of addon could even be enabled by default, but we should also consider making a "bug fixes" group to tuck away addons that 99% of users will want to keep on. (Related: #2617)

**Comment by Samq64** [comment]
For my stuff, account settings and unshared projects the page might not be accessible after loggging out.

**Reply by BroJac5246** [reply]
In those cases, we can default to going to the homepage.

**Reply by DNin01** [reply]
I believe those pages display a sign-in form when a user is logged out.

**Reply by BroJac5246** [reply]
> I believe those pages display a sign-in form when a user is logged out.

Yeah but it's not very useful to see that immediately after you log out. If we want, it could be a setting, though, if anyone does want to immediately see the sign-in screen.

**Reply by DNin01** [reply]
I think we might need to clear things up. If I'm right, the reason this suggestion was created was because sometimes you can land on the My Stuff page but be signed out. This could happen due to glitches where you are signed in, click a link to My Stuff, and then are suddenly signed out after going to My Stuff (given the timing of this suggestion, this is somewhat probable), or getting there from a browser bookmark or something.

From the discussion author's comment, it sounds like whenever you sign into Scratch through the form on the My Stuff page, you are redirected to the home page. Is that right?

In the meantime, I think you can work around this redirect by using the "Sign in" button in the navigation bar instead, as it consistently keeps you on the same page after signing in.

**Reply by BroJac5246** [reply]
Ah! I misunderstood the suggestion.
